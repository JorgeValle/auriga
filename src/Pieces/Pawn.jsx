import React from 'react';

// Components should represent one piece of data model
class Pawn extends React.Component {

  render() {
  
    return (
      <i class="fas fa-chess-pawn"></i>
    );
  }
}

export default Pawn;