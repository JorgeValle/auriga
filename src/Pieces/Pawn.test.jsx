import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import Pawn from './Pawn.jsx';

configure({
  adapter: new Adapter()
});

describe('<Pawn />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<Pawn />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<Pawn />);
    expect(wrapper).toMatchSnapshot();
  });

});

