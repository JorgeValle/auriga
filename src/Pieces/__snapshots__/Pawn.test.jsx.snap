// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pawn /> matches snapshot 1`] = `ShallowWrapper {}`;
